x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -102,13 +102,26 @@ primaryExpression
     ;
 
 functionExpression
-    : functionName LP (ASTERISK | (booleanExpression (COMMA booleanExpression)*))? RP
+    : functionName LP (ASTERISK | (functionArgument (COMMA functionArgument)*))? RP
     ;
```

Comment:
This allows things like `fn({}, param, {})`. Is it ok? I suppose we'll check it later, but I wonder if we should make the parsing strict instead. Something like this:
```suggestion
    : functionName LP (ASTERISK | map | (functionArgument (COMMA functionArgument)* (COMMA map)?))? RP
    ;

map
    : {this.isDevVersion()}? LEFT_BRACES mapExpression RIGHT_BRACES
    ;
```